### PR TITLE
Fix quickstart kind

### LINKED ISSUE #651: Quickstart with Kind is not working
I follow the Kind Quickstart (https://karina.docs.flanksource.com/admin-guide/provisioning/kind/). In step 5 (deploy the bare minimum configuration), I got some error log from karina cli

```
...
ERRO[0046] Failed to deploy base: cert-manager: failed to deploy cert-manager: failed to get dynamic client for &{map[apiVersion:cert-manager.io/v1 kind:ClusterIssuer metadata:map[name:default-issuer namespace:cert-manager] spec:map[ca:map[secretName:cert-manager-webhook-ca]]]}: no matches for kind "ClusterIssuer" in version "cert-manager.io/v1"
...
```

After the provision command, I checked pods in Kind cluster and noticed that `calico-node` is failing to start. Here is log from calico-pod:
```
2021-02-06 02:40:45.836 [INFO][10] startup.go 256: Early log level set to info
2021-02-06 02:40:45.839 [INFO][10] startup.go 272: Using NODENAME environment for node name
2021-02-06 02:40:45.839 [INFO][10] startup.go 284: Determined node name: test-cluster-control-plane
2021-02-06 02:40:45.840 [INFO][10] k8s.go 228: Using Calico IPAM
2021-02-06 02:40:45.840 [INFO][10] startup.go 316: Checking datastore connection
2021-02-06 02:40:45.848 [INFO][10] startup.go 340: Datastore connection verified
2021-02-06 02:40:45.849 [INFO][10] startup.go 95: Datastore is ready
2021-02-06 02:40:45.851 [INFO][10] customresource.go 100: Error getting resource Key=GlobalFelixConfig(name=CalicoVersion) Name="calicoversion" Resource="GlobalFelixConfigs" error=the server could not find the requested resource (get GlobalFelixConfigs.crd.projectcalico.org calicoversion)
2021-02-06 02:40:45.856 [INFO][10] startup.go 584: Using autodetected IPv4 address on interface eth0: 172.17.0.2/16
2021-02-06 02:40:45.856 [INFO][10] startup.go 647: No AS number configured on node resource, using global value
2021-02-06 02:40:45.857 [INFO][10] startup.go 149: Setting NetworkUnavailable to False
2021-02-06 02:40:45.877 [INFO][10] startup.go 536: CALICO_IPV4POOL_NAT_OUTGOING is true (defaulted) through environment variable
2021-02-06 02:40:45.877 [INFO][10] startup.go 796: Ensure default IPv4 pool is created. IPIP mode: Never, VXLAN mode: Never
2021-02-06 02:40:45.879 [ERROR][10] startup.go 802: Failed to create default IPv4 IP pool: 192.168.0.0/16 error=resource does not exist: IPPool(default-ipv4-ippool) with error: the server could not find the requested resource (post IPPools.crd.projectcalico.org)
2021-02-06 02:40:45.880 [WARNING][10] startup.go 1057: Terminating
Calico node failed to start
stream closed
```

All other pods stuck in pending state as calico couldn't setup the CNI.

But if I ignore those errors and continue to do step 6 (`karina deploy all -c test-cluster.yml`) then things start to come up healthy. After step 6 is finished, there are still 2 workloads are not happy which are: dex and platform-operator with `ImagePullBackOff` error.

The image for dex is `quay.io/dexidp/dex:v2.27.0` and for platform-operator is `docker.io/flanksource/platform-operator:0.5.1`.

**Is any of these errors expected? And what is the expected outcome for this Kind Quickstart?**

---

Karina version: `-20210202164958`
Kind version: `kind v0.9.0 go1.15.2 darwin/amd64`

### PR DESCRIPTION
**Pull request by tobernguyen**
Changes:
- fix: correct the default image tag of `platform-operator` and update to the latest version (`v0.5.2`)
- fix: change dex docker image repo to `ghcr.io/dexidp/dex`
- fix: update Kind Quickstart guide

What did I test:
- Built the latest binary from this branch
- Follow the updated Kind Quickstart guide from this branch
- Confirm all pods in the clusters are started up in healthy state

image

TODO:
- Will need to update the Kind Quickstart again once a new version is released after this PR is merged because the version in the example will not provision a healthy cluster.

Related PRs that can be closed after this is merged: https://github.com/flanksource/karina/pull/663 and https://github.com/flanksource/karina/pull/652

Fixes #651

### CODE REVIEW

File: docs/admin-guide/provisioning/kind.md [hunk 1]
```diff
@@ -4,21 +4,22 @@
 
 Download the latest official binary release for your platform from the [github repository](https://github.com/flanksource/karina/releases/latest).
 
-=== "Linux"
+#### Linux
 ```bash
-wget https://github.com/flanksource/karina/releases/download/v0.15.1/karina
+wget https://github.com/flanksource/karina/releases/download/v0.24.4/karina
 chmod +x karina
```
Comment (moshloop) [inline docs/admin-guide/provisioning/kind.md:10]:
```suggestion
wget -nv -nc -O karina https://github.com/flanksource/karina/releases/latest/download/karina && chmod +x karina
```

!!! info
    For production pipelines you should always pin the version of karina you are using

File: docs/admin-guide/provisioning/kind.md [hunk 2]
```diff
@@ -52,31 +53,39 @@ karina ca generate --name ingress-ca --cert-path .certs/ingress-ca.crt --private
...
 
 ```shell
 karina deploy all
 ```
 
+### Cleanup
+Stop and delete the container running Kind with
+```shell
+docker stop test-cluster-control-plane
+docker rm test-cluster-control-plane
```
Comment (moshloop) [inline docs/admin-guide/provisioning/kind.md:80]:
```suggestion
kind delete cluster --name test-cluster-control-plane
```

File: docs/admin-guide/provisioning/kind.md [hunk 3]
```diff
@@ -52,31 +53,39 @@ karina ca generate --name ingress-ca --cert-path .certs/ingress-ca.crt --private
...
+```
 
 
 ## Troubleshooting
 
 KIND cluster creation issues can be debugged by specifying the `--trace` argument to `karina` during creation:
 
-```bash
+```Shell
```
Comment (moshloop) [inline docs/admin-guide/provisioning/kind.md:88]: is this intentional?

Comment (tobernguyen) [reply docs/admin-guide/provisioning/kind.md:88]: no, it shouldn't be capitalized, pushing a fix

Comment (moshloop) [reply docs/admin-guide/provisioning/kind.md:88]: Whats the difference between bash vs shell in mkdocs?

Comment (tobernguyen) [reply docs/admin-guide/provisioning/kind.md:88]: I didn't think about mkdocs when updated this. I just saw all code boxes above are using `shell`. I usually zsh so it's a bit nicer in markdown but I'm not sure about mkdocs. Just looked at what was generated by mkdocs (I assume) here, it doesn't look different between `shell` and `bash`: https://karina.docs.flanksource.com/admin-guide/provisioning/kind/

File: docs/admin-guide/provisioning/kind.md (hunk 1 above)
Comment (tobernguyen) [reply docs/admin-guide/provisioning/kind.md]: Nice!

**Pull request merged**